### PR TITLE
Fix 64 bit install

### PR DESCRIPTION
**Pull request by Nicop06**
This patch will install libraries in lib64 instead of lib when doing a 64 bits
build.

### CODE REVIEW

**Comment by naelstrof** [comment]
I like it, thanks!

**Comment by Nicop06** [comment]
To give some credits, this was a fix for gentoo build (see <https://bugs.gentoo.org/629440>).

Especially, there was a link in this bug report containing the code I added: <https://cmake.org/pipermail/cmake/2013-July/055374.html>

File: CMakeLists.txt [hunk 1]
```diff
@@ -118,7 +118,15 @@ else()
   endif()
 endif()
 
+get_property(LIB64 GLOBAL PROPERTY FIND_LIBRARY_USE_LIB64_PATHS)
```
Comment (mgorny) [inline CMakeLists.txt]: This is horribly wrong. It's a very bad idea to rely on some random CMake internals to guess this, and even if it solves anything, it solves only 1 out of 5 or 6 different scenarios.

File: CMakeLists.txt [hunk 2]
```diff
@@ -118,7 +118,15 @@ else()
...
+
+if ("${LIB64}" STREQUAL "TRUE")
+  set(LIBSUFFIX 64)
```
Comment (mgorny) [inline CMakeLists.txt]: A name more commonly used in different CMake packages is `LIB_SUFFIX`, and it needs to be configurable. Alternatively, CMake has a built-in `GNUInstallDirs` package that handles paths correctly for different systems and guarantees portability, unlike hardcoding paths.

Comment (naelstrof) [reply CMakeLists.txt]: I'll take a better look at it when I'm not bogged by homework. There's a reason I haven't tagged a new release. If you wouldn't mind fixing it up and making a merge request that'd be appreciated.

**Comment by Nicop06** [comment]
@mgorny  Thank you for the tip,I should have looked further before submitting.

I updated the PR using the `GNUInstallDirs` module and the lib is correctly installed in /usr/lib64 instead of /usr/lib.

**Comment by bearcatsandor** [comment]
Is there a reason this hasn't been merged and released yet?

**Comment by SoapGentoo** [comment]
@Nicop06 Please see #93 for the proper fix with `GNUInstallDirs`, please some other minor niggles, like using ISO C++ over GNU C++, not compressing manpages etc.

**Comment by AloisMahdal** [comment]
Not to be a nitpick, but what about @mgorny's warning about line 121?  I admit I don't understand CMake so I actually don't know what to think of it, it's just that "horribly wrong " looks serious and I can't see it addressed here.

**Comment by SoapGentoo** [comment]
1. It uses an internal CMake property, and relying on implementation details is always bad.
2. My PR's use of `GNUInstallDirs` is much better, because it only uses a public interface, and `CMAKE_INSTALL_LIBDIR` is set up by default to point to either `lib`, `lib32` or `lib64`, depending on CHOST, distro etc. It also overridable cleanly (like the GNU conventions require).

**Comment by mgorny** [comment]
Also note that depending on the CPU, ABI and the system libdir values vary between `lib`, `lib32`, `lib64` and `libx32`. And there is no reason not to expect new values in the future.

**Comment by Nicop06** [comment]
@SoapGentoo Thank you for the link.

Indeed, this is exactly the same changes I did in the later version of the patch plus some unrelated changes. This is a solved issue and the master branch should now include the changes necessary to correctly configure the lib path, let it be `lib`, `lib32`, `lib64`, `libx32` or `libqubit`.

**Comment by AloisMahdal** [comment]
Thank you for explanation!

**Pull request merged**